Libraries/Database/SessionExtensions.cs

```diff
@@ -40,18 +40,53 @@ public static class SessionExtensions
                 parameters,
                 out var query,
                 out var messages);
-
-            if (!parserResult)
-            {
-                throw new RawQueryException(
-                    $"Error during parsing the query \"{sql}\" with the following messages: {Environment.NewLine}{string.Join(Environment.NewLine, messages)}",
-                    messages);
-            }
+            ThrowIfParseFailed(parserResult, sql, messages);
 
             return queryExecutor == null
                 ? await transaction.Connection.QueryAsync<TRow>(query, transaction: transaction)
                 : await queryExecutor((query, transaction));
         }
+
+        /// <summary>
+        /// Executes a raw SQL string command in a database-agnostic way by running it through Orchard's <see
+        /// cref="SqlParser"/>.
+        /// </summary>
+        /// <param name="sql">
+        /// The raw SQL string. Doesn't need to use table prefixes or care about SQL dialects.
+        /// </param>
+        /// <param name="parameters">Input parameters passed to the query.</param>
+        /// <param name="transaction">If not <see langword="null"/> it must be an open DB transaction.</param>
+        /// <returns>The number of rows affected.</returns>
+        public static async Task<int> RawExecuteAsync(
+            this ISession session,
+            string sql,
+            IDictionary<string, object> parameters = null,
+            DbTransaction transaction = null)

```

Comment:
Szerintem inkább legyen ugyanúgy paraméterezve, mint a `RawQueryAsync()`, csak tényleg a visszatérési értékben különbözzön. Ha használhatja esetleg azt egyből, akkor még jobb.

Reply:
> Szerintem inkább legyen ugyanúgy paraméterezve, mint a RawQueryAsync(),

Oda is átmásolom a `transaction` paramétert. A `queryExecutor`-nak itt nem látom értelmét mivel nem ad vissza adatot.

> Ha használhatja esetleg azt egyből, akkor még jobb.

Nem tudja, mivel pont az utolsó metódushívás a különböző (`Connection.QueryAsync<T>()` vs `Connection.ExecuteAsync()`) De még jobban összevontam.